src/libraries/Microsoft.Extensions.Caching.Memory/ref/Microsoft.Extensions.Caching.Memory.cs

```diff
@@ -45,6 +46,7 @@ public partial class MemoryCacheOptions : Microsoft.Extensions.Options.IOptions<
         Microsoft.Extensions.Caching.Memory.MemoryCacheOptions Microsoft.Extensions.Options.IOptions<Microsoft.Extensions.Caching.Memory.MemoryCacheOptions>.Value { get { throw null; } }
         public long? SizeLimit { get { throw null; } set { } }
         public bool TrackLinkedCacheEntries { get { throw null; } set { } }
+        public bool TrackStatistics { get { throw null; } set { } }
```

Comment:
what do people think about making the default value of this property true? I suspect most developers, even those writing high performance services, will still probably prefer being able to monitor their cache statistics rather than squeezing out a tiny bit more speed.

Reply:
during the API review meeting today the ask was to disable it by default to not hurt the perf for those not tracking the stats

Reply:
Just tested it again now, with TrackStatistics enabled. 

As summarized earlier, the new approach would drop the TE RPS by 5%:

![image](https://user-images.githubusercontent.com/5897654/159603106-fc2120a5-d02b-466b-8d2b-b13b66ae6336.png)

Reply:
IMO true would be the better default but I'm not going to die on that hill : )

Reply:
We can disable stats in the benchmark though, this is similar to what we do with logging.  That should apply here as well right?

Reply:
I think the difference is that far more people want logging by default than will want `MemoryCache.GetCurrentStatistics()` by default. On the other hand as @eerhardt said in API review, this benchmark is pretty much the worst-case overhead.

I think defaulting to not tracking makes sense, but I'm not super opinionated.

Reply:
I'd be fine either way, though this would need another API review churn. Would prefer to have this PR merged as-is and park this discussion on a separate issue if OK.

Reply:
This would be my speculation how projects break down:

1. Projects that deploy on x86 and run within one order of magnitude of the load created by the TE benchmark: < 0.1% of apps
2. Projects where the developer wants to directly invoke GetCurrentStatistics(): <1% of apps
3. Projects where the developer will appreciate using a monitoring solution (AppInsights, OpenTelemetry, etc) or ad-hoc tools (dotnet-counters) to make the stats available for them in the future: 10-20% apps
4. Projects where neither the stats nor the perf matters: 50+% of apps

I'm worried about the experience of people in group (3). I want them to be able to turn on telemetry very easily when they want it. Anything where they have to follow extra instructions or change more lines of code creates friction in that process. If that means a tiny set of projects in group (1) have to have some friction to disable stats they don't care about I think it is the right tradeoff given my guess at the relative sizes of those groups. Certainly if others think my guesses about the group sizes are wrong they might come to a different conclusion.

Reply:
I'd assume for case 3, when using dotnet-counters there would still be need for code change for setting up counters with the GetCurrentStatistics API and in which case also setting the flag should be not too much churn.

But thinking perhaps with OT or AppInsights there is no need for code change and therefore, as you said once those solutions are in for cache metrics retrieval via OT, then it could be that extra instruction for them to also set TrackStatistics to true.